torchvision/prototype/datasets/utils/_internal.py

```diff
@@ -250,3 +255,46 @@ def _make_sharded_datapipe(root: str, dataset_size: int) -> IterDataPipe:
     # dp = dp.cycle(2)
     dp = TakerDataPipe(dp, dataset_size)
     return dp
+
+
+def fromfile(
+    file: BinaryIO,
+    *,
+    dtype: torch.dtype,
+    byte_order: str,
+    count: int = -1,
+) -> torch.Tensor:
+    """Construct a tensor from a binary file.
+
+    .. note::
+
+        This function is similar to :func:`numpy.fromfile` with two notable differences:
+
+        1. This function only accepts an open binary file, but not a path to it.
+        2. This function has an additional ``byte_order`` parameter, since PyTorch's ``dtype``'s do not support that
+            concept.
+
+    Args:
+        file (IO): Open binary file.
+        dtype (torch.dtype): Data type of the returned tensor.
+        byte_order (str): Byte order of the data. Can be ``"little"`` or ``"big"`` endian.
+        count (int): Number of values of the returned tensor. If ``-1`` (default), will read the complete file.
+    """
+    byteorder = "<" if byte_order == "little" else ">"
+    char = "f" if dtype.is_floating_point else ("i" if dtype.is_signed else "u")
+    itemsize = (torch.finfo if dtype.is_floating_point else torch.iinfo)(dtype).bits // 8
+    np_dtype = byteorder + char + str(itemsize)
+
+    buffer = file.read(-1 if count == -1 else count * itemsize)
+    # torch.frombuffer can only deal with with the native byte order,
+    # so we use numpy for the I/O and convert to a tensor.
+    return torch.from_numpy(np.frombuffer(buffer, dtype=np_dtype).astype(np_dtype[1:]))
+
+
+def read_flo(file: BinaryIO) -> torch.Tensor:
+    if file.read(4) != b"PIEH":
+        raise ValueError("Magic number incorrect. Invalid .flo file")
+
+    width, height = fromfile(file, dtype=torch.int32, byte_order="little", count=2)
```

Comment:
AFAIK, byte order is foreign to PyTorch. Thus, there is no reference to go on. In general though, PyTorch's policy is to side with Python over `numpy` in case they disagree. In Python [`sys.byteorder`](https://docs.python.org/3/library/sys.html#sys.byteorder) is either `"little"` or `"big"`. Plus, although more verbose, I find that more expressive than using `">"` and `"<"`.